octavia_f5/controller/worker/controller_worker.py

```diff
@@ -850,3 +853,90 @@ def register_in_availability_zone(self, az_name):
                     'availability_zone_profile_id': azp_dict['id']
                 }
                 self._az_repo.create(lock_session, **az_dict)
+
+
+class ControllerWorkerNotifications(object):
+
+    def __init__(self):
+        self._loadbalancer_repo = f5_repos.LoadBalancerRepository()
+        self._vip_repo = repo.VipRepository()
+
+        # instantiate managers/drivers
+        self.network_driver = driver_utils.get_network_driver()
+
+        # RPC client to send notifications to another workers
+        self.target = messaging.Target(
+            namespace=octavia_consts.RPC_NAMESPACE_CONTROLLER_AGENT,
+            topic=octavia_consts.TOPIC_AMPHORA_V2, version="2.0", fanout=False)
+        self.rpc_client = rpc.get_client(self.target)
+
+        super().__init__()
+
+    def _get_server(self, loadbalancer):
+        """ Get scheduled host of the loadbalancer.
+        :param loadbalancer: loadbalancer data
+        :return: scheduled host
+        """
+        if loadbalancer.server_group_id:
+            return loadbalancer.server_group_id
+        # fetch scheduled server from VIP port
+        return self.network_driver.get_scheduled_host(loadbalancer.vip.port_id)
+
+    def _get_all_sgs(self, security_group):
+        found_sgs = [security_group]
+        rules = list(tuple(self.network_driver.network_proxy.security_group_rules(
+            security_group_id=security_group)))
+        for rule in rules:
+            if rule.get('remote_group_id'):
+                found_sgs += self._get_all_sgs(rule['remote_group_id'])
+        return found_sgs
+
+    def process_security_group_update_notification(self, security_group_id, action):
```

Comment:
Note to myself: How can we guarantee that this is only invoked on one worker? Wait, it's a normal method call in `octavia/controller/queue/v2/endpoints.py`. Does this mean this runs in the Octavia API process?